clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1524,11 +1534,29 @@ String coordinatorKey() {
         @Override
         public void handleResponse(AbstractResponse response) {
             FindCoordinatorResponse findCoordinatorResponse = (FindCoordinatorResponse) response;
-            Errors error = findCoordinatorResponse.error();
             CoordinatorType coordinatorType = CoordinatorType.forId(builder.data().keyType());
 
+            if (batchFindCoordinator && findCoordinatorResponse.data().coordinators().size() != 1) {
+                log.error("Group coordinator lookup failed: Invalid response");
+                fatalError(new IllegalStateException("Group coordinator lookup failed: Invalid response"));
```

Comment:
Could we have a more descriptive error message/exception here? Something more about how any coordinator size that's not 1 is unexpected since we are only looking up the coordinator for one group.